src/main/java/forestry/arboriculture/blocks/BlockWoodPile.java

```diff
@@ -201,7 +201,11 @@ private float getCharcoalAmount(World world, BlockPos pos) {
 		for (EnumFacing facing : EnumFacing.VALUES) {
 			charcoalAmount += getCharcoalFaceAmount(world, pos, facing);
 		}
-		return charcoalAmount / 6;
+		float ret = charcoalAmount / 6;
+		if (ret > 0F) {
+			ret += 9F;    //base charcoal drop amount for ash block, calculated here instead so invalid walls won't still drop charcoal
```

Comment:
Why is this at `BlockWoodPile.getCharcoalAmount` and not in `BlockAsh.getDrops` after `if (amount > 0)` ?

Reply:
Not sure, probably I got confused. Having it in this method changes behaviour too because the amount gets clamped at the end.